src/mainwindow.h

```diff
@@ -21,25 +24,30 @@ class MainWindow : public QMainWindow
     bool event(QEvent* event) override;
     void resizeEvent(QResizeEvent* event) override;
     ~MainWindow();
-    void setImage(std::string path);
+    void setImage(const ImageDetails_t &imageDetails);
     void setBlurRadius(unsigned int blurRadius);
     void setBackgroundOpacity(unsigned int opacity);
     void warn(std::string text);
     void setOverlay(Overlay* overlay);
-    void setAspect(char aspectIn);
     void setDebugMode(bool debugModeIn) {debugMode = debugModeIn;}
-    void setFitAspectAxisToWindow(bool fitAspectAxisToWindowIn) { fitAspectAxisToWindow = fitAspectAxisToWindowIn; }
+    void setBaseOptions(const ImageDisplayOptions_t &baseOptionsIn);
+    const ImageDisplayOptions_t &getBaseOptions() { return baseImageOptions; }
+    void setImageSwitcher(ImageSwitcher *switcherIn) { switcher = switcherIn; }
```

Comment:
please don't inline (sorry missed it in the preveious pr)

Reply:
Done and removed the other ones I've added (and found when I did a quick check)